Client/Frontend/Browser/BackForwardListView.swift

```diff
@@ -0,0 +1,196 @@
+            tableView.backgroundView = blurEffectView
+        }
+        else {
+            tableView.backgroundColor = UIColor.init(colorLiteralRed: 1, green: 1, blue: 1, alpha: 0.4)
+            let blurEffect = UIBlurEffect(style: .ExtraLight)
+            let blurEffectView = UIVisualEffectView(effect: blurEffect)
+            tableView.backgroundView = blurEffectView
+        }
+        
+        let sql = profile.favicons as! SQLiteHistory
+        var urls: [String] = [String]()
+        
+        for page in backForwardList.forwardList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Forward))
+        }
+        
+        var currentRow = 0
+        if let currentPage = backForwardList.currentItem {
+            currentRow = listData.count
+            urls.append(currentPage.URL.absoluteString)
+            listData.append((currentPage, .Current))
+        }
+        for page in backForwardList.backList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Backward))
+        }
+        
+        listData = listData.filter { !(($0.item.title ?? "").isEmpty)}
+        
+        let deferred = sql.getSitesForURLs(urls)
+        
+        deferred.uponQueue(dispatch_get_main_queue()) { result in
+            if let cursor = result.successValue {
+                for cursorSite in cursor {
+                    if let site = cursorSite, let url = site?.url {
+                        self.sites[url] = site
+                    }
+                }
+                self.tableView.reloadData()
+            }
+        }
+        
+        addSubview(tableView)
+        tableView.snp_makeConstraints { make in
+            make.height.equalTo(0)
+            make.left.right.bottom.equalTo(self)
+        }
+        
+        if currentRow > 1 {
+            self.tableView.reloadData()
+            let moveToIndexPath = NSIndexPath(forRow: currentRow-2, inSection: 0)
+            self.tableView.reloadRowsAtIndexPaths([moveToIndexPath], withRowAnimation: .None)
+            self.tableView.scrollToRowAtIndexPath(moveToIndexPath, atScrollPosition: UITableViewScrollPosition.Middle, animated: false)
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        if dismissing {
+            return
+        }
+        
+        let height = min(CGFloat(BackForwardViewUX.RowHeight*listData.count), self.frame.height/2)
+        UIView.animateWithDuration(0.2, animations: {
+            self.tableView.snp_updateConstraints { make in
+                make.height.equalTo(height)
+            }
+            self.backgroundColor = UIColor.init(colorLiteralRed: 0, green: 0, blue: 0, alpha: 0.2)
+            self.layoutIfNeeded()
+        })
```

Comment:
There shouldn't be any animation calls inside a `layoutSubviews` since this method can get call on every rotation and layout validation. We should look into alternative ways of animating the height of the `tableView`.